Sources/Node.swift

```diff
@@ -3,10 +3,22 @@ import Foundation
 
 public struct TemplateSyntaxError : Error, Equatable, CustomStringConvertible {
   public let description:String
+  public var token: Token?
 
   public init(_ description:String) {
     self.description = description
   }
+  
+  public func contextAwareError(templateName: String?, templateContent: String) -> TemplateSyntaxError? {
```

Comment:
I'd prefer to keep this logic out of the error class itself. I'd keep `TemplateSyntaxError` as a [POO](https://en.wikipedia.org/wiki/Plain_old_Java_object) without additional logic included.

I'd prefer to see the error containing all of the stack frames and contextual information and a separate renderer can be capable of rendering it in different ways.

There will be separate renders for ANSI, plain text, HTML, Bugsnag reporter, etc (eventually).